stan/math/opencl/gp_exp_quad_cov.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef STAN_MATH_OPENCL_GP_EXP_QUAD_COV_HPP
+#define STAN_MATH_OPENCL_GP_EXP_QUAD_COV_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/kernels/gp_exp_quad_cov.hpp>
+#include <stan/math/opencl/err/check_matching_dims.hpp>
+#include <CL/cl.hpp>
+
+namespace stan {
+namespace math {
+/**
+ * Squared exponential kernel on the GPU.
+ *
+ * @param x input vector or matrix
+ * @param sigma standard deviation
+ * @param length_scale length scale
+ *
+ * @return Squared distance between elements of x.
+ */
+inline matrix_cl gp_exp_quad_cov(const matrix_cl& x, const double sigma,
+                                 const double length_scale) try {
+  matrix_cl res(x.cols(), x.cols());
+  opencl_kernels::gp_exp_quad_cov(cl::NDRange(x.cols(), x.cols()), x, res,
+                                  sigma * sigma, -0.5 / square(length_scale),
+                                  x.cols(), x.rows());
+  return res;
+} catch (const cl::Error& e) {
+  check_opencl_error("gp_exp_quad_cov", e);
+  // check above causes termination so below will not happen
```

Comment:
why do you need this comment? If that's the case, why even bother with the rest of the code?

Reply:
> why do you need this comment? If that's the case, why even bother with the rest of the code?

It's not that it's needed, as you said before we don't normally use try functions so for people reading I was like, "Hey fyi this does not return"